.github/workflows/update-docs.yml

```
@@ -0,0 +1,45 @@
+name: Update Docs
+on:
+  pull_request:
+
+jobs:
+  update_dfx_json_schema:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        rust: ["1.60.0"]
+        os: [ ubuntu-20.04 ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Install Rust
+        run: |
+          rustup update ${{ matrix.rust }} --no-self-update
+          rustup component add clippy
+          rustup default ${{ matrix.rust }}
+
+      - name: Check cargo build
+        run: cargo build
+      - name: Show download worked
+        run: cargo run -- --version
+      - name: Update docs/dfx-json-schema.json
+        run: |
+          cargo run -- schema --outfile docs/dfx-json-schema.json
+          cat docs/dfx-json-schema.json
+          if ! git diff-index --quiet HEAD; then
```

```suggestion
          if ! git diff-index --quiet HEAD -- docs/dfx-json-schema.json; then
```
have you tried this one?